File: aula3/java_logic/calculadora.java
import java.util.Scanner; // Importa a classe para leitura de dados [cite: 11]
import java.util.Locale; // Importa para garantir o uso do ponto decimal em vez de vírgula

public class calculadora {
    public static void main(String[] args) {
        // Configura o ponto como separador decimal (padrão americano/programação)
        Locale.setDefault(Locale.US);

        // Passo 1: instanciar o objeto Scanner para ler a entrada do teclado [cite: 11]
        Scanner sc = new Scanner(System.in);

        //Passo 2: entrada de dados e declaração de variáveis [cite: 6]
        System.out.println(x: "--- Calculadora dde Negócios (Java) ---");

        System.out.print(s: "Digite o primeiro número: ");
        double numero1 = sc.nextDouble(); // Java requer definição explícita do tipo [cite: 6]

        System.out.print(s: "Digite o segundo número: ");
        double numero2 = sc.nextDouble();

        //Passo 3: processamento (cálculos aritméticos) [cite: 7]
        double soma = numero1 + numero2;
        double subtracao = numero1 - numero2;
        double multiplicao = numero1 * numero2;
        double divisao = numero1 / numero2;

        //Passo 4: saída de resultados formatados
        System.out.printf(format: "Soma: %.2f%n", soma);
        System.out.printf(format: "Subtracao: %.2f%n", subtracao);
        System.out.printf(format: "Multiplicação: %.2f%n", multiplicao);
        System.out.printf(format: "Divisão: %.2f%n", divisao);

        // Passo 5: fechar o Scanner (boa prática de gerenciamento de memória)
        sc.close();
    }
}

File: Aula 03 - Variaveis, Tipos de Dados e Operadores/cpp_logic/calculadora.cpp
#include <iostream> // Biblioteca para entrada e saída de dados
#include <iomanip> // Biblioteca para formatar a saída de casas decimais
using namespace std;

int main() {
    // Passo 1: declaração de variáveis (Tipagem Estática)
    int numero1 = 0, numero2 = 0, soma, subtracao, multiplicacao;
    float divisao;

    // Passo 2: entrada de dados
    cout << "Digite o primeiro número: "; cin >> numero1;
    cout << "Digite o segundo numero: "; cin >> numero2;

    // Passo 3: processamento (Cálculos Aritmeticos)
    soma = numero1 + numero2;
    subtracao = numero1 - numero2;
    multiplicacao = numero1 * numero2;
    divisao = numero1 / numero2;

    //Passo 4: saída de resultados formatados
    cout << fixed << setprecision(2);
    cout << "Soma: " << soma << endl;
    cout << "Subtracao: " << subtracao << endl;
    cout << "Multiplicação: " << multiplicacao << endl;
    cout << "Divisão: " << divisao << endl;

    return 0; // Indica que o programa terminou com sucesso.


}